feat(codemod): codemod PermissionOverwrite typings as well

onerandomusername opened the pull request:
## Summary

codemod the final part of permission typings :)

## Checklist

<!-- Put an x inside [ ] to check it, like so: [x] -->

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

shiftinv commented:
Could you add a changelog entry for the now fixed `use_application_commands` and `use_embedded_activities` annotations?

onerandomusername commented:
> Could you add a changelog entry for the now fixed `use_application_commands` and `use_embedded_activities` annotations?

Didn't notice those were missing, fixed in https://github.com/DisnakeDev/disnake/pull/777/commits/6c86cd2b9fd84f53d6863f1fd8befe4722420132
